pkg/platform/runtime/runtime.go

```diff
@@ -20,15 +23,20 @@ type Runtime struct {
 	msgHandler  MessageHandler
 }
 
-func NewRuntime(commitID strfmt.UUID, owner string, projectName string, msgHandler MessageHandler) *Runtime {
-	var installPath string
-	if runtime.GOOS == "darwin" {
-		// mac doesn't use relocation so we can safely use a longer path
-		installPath = filepath.Join(config.CachePath(), owner, projectName)
-	} else {
-		installPath = filepath.Join(config.CachePath(), hash.ShortHash(owner, projectName))
+func NewRuntime(projectDir string, commitID strfmt.UUID, owner string, projectName string, msgHandler MessageHandler) (*Runtime, error) {
```

Comment:
`projectDir` should probably be `cacheDir`

Reply:
I think `projectDir` is a better name here, because it actually points to the directory where the `activestate.yaml` file is located.  We create the `cacheDir` from that `projectDir` (by hashing it), as the AC requested.

Reply:
ahh gotcha